src/ctapipe/io/dl2_tables_preprocessing.py

```diff
@@ -0,0 +1,474 @@
+            Column(name="reco_alt", unit=u.deg, description="Reconstructed altitude"),
+            Column(name="pointing_az", unit=u.deg, description="Pointing azimuth"),
+            Column(name="pointing_alt", unit=u.deg, description="Pointing altitude"),
+        ],
+        help="Schema definition for output event QTable",
+    ).tag(config=True)
+
+    def __init__(self, config=None, parent=None, **kwargs):
+        super().__init__(config=config, parent=parent, **kwargs)
+        self.quality_query = DL2EventQualityQuery(parent=self)
+
+    @default("columns_to_rename")
+    def _default_columns_to_rename(self):
+        return {
+            f"{self.energy_reconstructor}_energy": "reco_energy",
+            f"{self.geometry_reconstructor}_az": "reco_az",
+            f"{self.geometry_reconstructor}_alt": "reco_alt",
+            f"{self.gammaness_classifier}_prediction": "gh_score",
+            "subarray_pointing_lat": "pointing_alt",
+            "subarray_pointing_lon": "pointing_az",
+        }
+
+    def normalise_column_names(self, events: QTable) -> QTable:
+        """
+        Rename column names according to configuration.
+
+        Parameters
+        ----------
+        events : QTable
+            Input event table.
+
+        Returns
+        -------
+        QTable
+            Table with selected and renamed columns.
+
+        Raises
+        ------
+        NotImplementedError
+            If pointing is not AltAz or varies too much.
+        ValueError
+            If required columns are missing.
+        """
+        if self.apply_check_pointing:
+            if events["subarray_pointing_lat"].std() > 1e-3:
+                raise NotImplementedError(
+                    "No support for making irfs from varying pointings yet"
+                )
+            if any(
+                events["subarray_pointing_frame"] != CoordinateFrameType.ALTAZ.value
+            ):
+                raise NotImplementedError(
+                    "At the moment only pointing in altaz is supported."
+                )
+
+        columns_to_keep = [col.name for col in self.output_table_schema]
+
+        rename_dict = self.columns_to_rename
+        rename_from = list(rename_dict.keys())
+        rename_to = list(rename_dict.values())
+
+        fixed_columns = list(set(columns_to_keep) - set(rename_to))
+        columns_to_read = fixed_columns + rename_from
+        for col in columns_to_read:
+            if col not in events.colnames:
+                raise ValueError(
+                    f"Input files must conform to the ctapipe DL2 data model. "
+                    f"Required column {col} is missing."
+                )
+
+        events = QTable(events[columns_to_read], copy=COPY_IF_NEEDED)
+        if rename_from and rename_to:
+            events.rename_columns(rename_from, rename_to)
+        return events
+
+    def make_empty_table(self) -> QTable:
```

Comment:
The only place where I could find this method being used is in a test fixture (`irf_events_table()`), and even there it's not really used directly, but the result is modified.  Maybe we should just remove this method and put what is needed in the fixture? Or does it have some other use?